pkg/services/ngalert/api/api_convert_prometheus_test.go

```diff
@@ -420,9 +494,211 @@ func TestRouteConvertPrometheusGetRules(t *testing.T) {
+			ruleStore.PutRule(context.Background(), rule2)
+			err := provenanceStore.SetProvenance(context.Background(), rule2, 1, models.ProvenanceAPI)
+			require.NoError(t, err)
+
+			response := srv.RouteConvertPrometheusDeleteNamespace(rc, fldr.Title)
+			require.Equal(t, http.StatusConflict, response.Status())
+
+			// Verify the rule is still present
+			remaining, err := ruleStore.GetAlertRuleByUID(context.Background(), &models.GetAlertRuleByUIDQuery{
+				UID:   rule.UID,
+				OrgID: rule.OrgID,
+			})
+			require.NoError(t, err)
+			require.NotNil(t, remaining)
+		})
+	})
+}
+
+func TestRouteConvertPrometheusDeleteRuleGroup(t *testing.T) {
+	t.Run("for non-existent folder should return 404", func(t *testing.T) {
+		srv, _, _, _ := createConvertPrometheusSrv(t)
+		rc := createRequestCtx()
+
+		response := srv.RouteConvertPrometheusDeleteRuleGroup(rc, "non-existent", "test-group")
+		require.Equal(t, http.StatusNotFound, response.Status())
+	})
+
+	const groupName = "test-group"
+
+	t.Run("valid request should delete rules", func(t *testing.T) {
+		initGroup := func(promDefinition string, groupName string, opts ...convertPrometheusSrvOptionsFunc) (*ConvertPrometheusSrv, *fakes.RuleStore, *folder.Folder, *models.AlertRule) {
+			srv, _, ruleStore, folderService := createConvertPrometheusSrv(t, opts...)
+
+			// Create a folder in the root
+			fldr := randFolder()
+			fldr.ParentUID = ""
+			folderService.ExpectedFolder = fldr
+			folderService.ExpectedFolders = []*folder.Folder{fldr}
+			ruleStore.Folders[1] = append(ruleStore.Folders[1], fldr)
+
+			rule := models.RuleGen.
+				With(models.RuleGen.WithNamespaceUID(fldr.UID)).
+				With(models.RuleGen.WithOrgID(1)).
+				With(models.RuleGen.WithGroupName(groupName)).
+				With(models.RuleGen.WithPrometheusOriginalRuleDefinition(promDefinition)).
+				GenerateRef()
+			ruleStore.PutRule(context.Background(), rule)
+
+			return srv, ruleStore, fldr, rule
+		}
+
+		t.Run("valid request should delete rules", func(t *testing.T) {
+			srv, ruleStore, fldr, rule := initGroup("prometheus definition", groupName)
+			rc := createRequestCtx()
+
+			// Create another rule in a different group that should not be deleted
+			otherGroupName := "other-group"
+			otherRule := models.RuleGen.
+				With(models.RuleGen.WithNamespaceUID(fldr.UID)).
+				With(models.RuleGen.WithOrgID(1)).
+				With(models.RuleGen.WithGroupName(otherGroupName)).
+				With(models.RuleGen.WithPrometheusOriginalRuleDefinition("other prometheus definition")).
+				GenerateRef()
+			ruleStore.PutRule(context.Background(), otherRule)
+
+			response := srv.RouteConvertPrometheusDeleteRuleGroup(rc, fldr.Title, groupName)
+			require.Equal(t, http.StatusAccepted, response.Status())
+
+			// Verify the rule was deleted
+			remaining, err := ruleStore.GetAlertRuleByUID(context.Background(), &models.GetAlertRuleByUIDQuery{
+				UID:   rule.UID,
+				OrgID: rule.OrgID,
+			})
+			require.Error(t, err)
+			require.Nil(t, remaining)
+		})
```

Comment:
Do another quick check here that the `other-group` is still there?

Reply:
Oh, right :)